tests/loader_layer_tests.cpp

```diff
@@ -2090,6 +2091,80 @@ TEST(ExplicitLayers, WrapObjects) {
     }
 }
 
+TEST(ExplicitLayers, VkLayerPathEnvVar) {
+    FrameworkEnvironment env;
+    env.add_icd(TestICDDetails(TEST_ICD_PATH_VERSION_2_EXPORT_ICD_GPDPA));
+    env.get_test_icd().add_physical_device({});
+
+    { 
+        // verify layer loads successfully when setting VK_LAYER_PATH to a full filepath
+        const char* regular_layer_name_1 = "RegularLayer1";
+        env.add_explicit_layer(
+        TestLayerDetails(
+            ManifestLayer{}.add_layer(
+                ManifestLayer::LayerDescription{}.set_name(regular_layer_name_1).set_lib_path(TEST_LAYER_PATH_EXPORT_VERSION_2)),
+            "regular_layer_1.json").set_discovery_type(ManifestDiscoveryType::env_var).set_is_dir(false));
+
+        InstWrapper inst(env.vulkan_functions);
+        inst.create_info.add_layer(regular_layer_name_1);
+        inst.CheckCreate(VK_SUCCESS);
+        auto phys_dev = inst.GetPhysDev();
+        uint32_t count = 0;
+        env.vulkan_functions.vkEnumerateDeviceLayerProperties(phys_dev, &count, nullptr);
+        ASSERT_EQ(count, 1U);
```

Comment:
Not a bad idea to check that the names returns match what you expect. When dealing with 'multiple' layers, there is a utility function `check_permutation` which verifies that all layer names are present in the list of extensions.

Reply:
@charles-lunarg added those extra verifications